### PR TITLE
feat: add fcc-claude-desktop launcher and wire desktop startup

### PR DESCRIPTION
**Pull request by Gakpey**
Part 6 of 7 — splits the closed #1545 into reviewable pieces (≤750 lines each).
Each part is a separate PR against `main` and should be merged in order (after Part 5); until earlier parts merge, this diff against `main` also shows them.

## What this PR does

Adds the `fcc-claude-desktop` launcher and wires the desktop shell to the routing layer.

- New `fcc-claude-desktop` console script: a thin shim over the config merge/launch helpers so install scripts and users can configure, unconfigure, or spawn Claude Desktop directly.
- `launch_desktop` starts the TLS front proxy and merges the Claude Desktop config before running the server, stopping the proxy on exit.
- The tray gains a "Launch Claude Desktop" item.
- Completes Linux desktop test coverage for the merge hook and tray launch item.

## Verification

- `ruff format --check`, `ruff check`, `ty check` clean
- Full pytest suite passes

<h3>Greptile Summary</h3>

The Claude Desktop integration adds launcher, routing configuration, verified HTTPS gateway management, and desktop-specific routing support. A writable pre-existing Caddy state directory can still be used to impersonate the trusted HTTPS gateway and capture the Desktop proxy credential. Separately, removing FCC can leave model discovery enabled when the user had previously disabled it.

<h3>Confidence Score: 2/5</h3>

The Desktop gateway trust boundary remains unsafe because attacker-controlled persisted Caddy state can be accepted as a trusted HTTPS front, and configuration removal does not consistently restore the user’s discovery preference.

The remaining findings include one credential-exposure issue and one configuration-restoration issue.

**Files Needing Attention:** src/free_claude_code/cli/tls_proxy.py, src/free_claude_code/cli/claude_desktop.py

<details open><summary><h3>Security Review</h3></summary>

A local user who can pre-populate a writable Caddy state directory can supply the front identity and managed CA material that FCC uses for gateway verification. FCC then accepts an attacker-controlled HTTPS listener, which receives Claude Desktop’s reusable proxy credential and inference traffic.
</details>

<details><summary><h3><a href="https://www.greptile.com/trex">T-Rex</a> T-Rex Logs</h3></summary>

**What T-Rex did**
- T-Rex produced a finding-comment-proof for a posted P1 finding and attached the focused harness source artifact to support reviewer inspection.
- T-Rex produced a second finding-comment-proof for a posted P1 finding.
- T-Rex performed general contract validation showing the vulnerable existing-identity path leaves the attacker-writable home, controlled listener adoption, and credential exposure to the adopted listener.

<a href="https://app.greptile.com/trex/runs/21761337/artifacts"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/ViewAllArtifactsDark.svg?v=4"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/ViewAllArtifacts.svg?v=4">View all artifacts</picture></a>

<sub><a href="https://www.greptile.com/trex">T-Rex</a> Ran code and verified through T-Rex</sub>
</details>

<h3>Comments Outside Diff (1)</h3>

1. General comment 

   <a href="#">P1</a> **Existing front identity bypasses Caddy-home permission hardening**

   - **Bug**
     - Under the documented other-local-user threat model, an attacker who can pre-populate an existing group/world-writable Caddy home can supply both `front-identity` and managed CA state. `load_or_create_front_identity` returns the attacker identity without restricting the home. A TLS listener certificate signed by the attacker-controlled managed CA then passes the ownership probe, is adopted as the HTTPS gateway, and receives client credentials.
   - **Cause**
     - `load_or_create_front_identity` reads and returns a non-empty existing identity at lines 143-149 before invoking `_restrict_home_permissions(path.parent)` at line 152. The trust anchor and identity can therefore be attacker-controlled when the directory was writable before the victim runs.
   - **Fix**
     - Restrict and validate the existing Caddy home before reading `front-identity` or managed CA state; reject unsafe ownership/permission states rather than trusting pre-existing contents. Add a regression test covering an existing non-empty identity in a `0777` home.

   <sub><a href="https://www.greptile.com/trex">T-Rex</a> Ran code and verified through T-Rex</sub>

<sub>Reviews (36): Last reviewed commit: ["fix: hand off a re-enabled discovery fla..."](https://github.com/alishahryar1/free-claude-code/commit/607423078670d5070ee93e32d6ca315cdc70bcd7) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=56608150)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: src/free_claude_code/cli/claude_desktop.py [hunk 1]
```diff
@@ -0,0 +1,559 @@
...
+    inference_dict: dict[str, object] = {}
+    if isinstance(inference_raw, dict):
+        for key, value in inference_raw.items():
+            inference_dict[str(key)] = value
+
+    if backup is not None:
+        # Snapshot path: the merge recorded what FCC replaced, so the
+        # discovery flag and every managed key are FCC's — remove them
+        # all, then restore the snapshot below.
+        if _DISCOVERY_KEY in data:
```
[the remark this answers is not in the document]
Comment (Gakpey) [reply src/free_claude_code/cli/claude_desktop.py]: Fixed in 1cd8f97 ("fix: absorb user edits when re-merging onto the desktop config snapshot", now the PR head).

Re-merges onto an object `inference` block now absorb the user's post-merge edits into the snapshot before the merge overwrites them: a managed key whose current value differs from what the previous merge wrote becomes the snapshot's new restore target, while keys still at their written value keep the original targets. The comparison is against the recorded written block (not a live re-resolution), so a rotated proxy token or gateway URL is not misread as a user edit. The discovery flag follows the same rule for any post-merge flip to a non-`True` value.

One caveat on the exact T-Rex repro: after the fix, the later user provider edit (`user-later-provider`) is preserved and the provider half of the repro passes. The discovery flag in that repro was flipped to `true` between merges, and a current `true` is byte-identical to configure's own always-`true` write — the file carries no state distinguishing "user set true" from "configure wrote true", so unconfigure keeps the recorded original there rather than guessing. If preserving user-set `true` is required, the flag would need a merge-sequence marker rather than a value comparison, which is a design change rather than a fix; happy to take that on if wanted.

Regression guards added alongside the fix: managed-key edit absorption, frozen snapshot without user edits, post-merge discovery flip to `false`, and the rotation-is-not-a-user-edit case.

File: src/free_claude_code/cli/desktop_entrypoint.py [hunk 2]
```diff
@@ -17,10 +19,21 @@ def launch(argv: Sequence[str] | None = None) -> None:
...
-    if sys.platform not in {"darwin", "win32"}:
-        print("FCC Desktop is supported on Windows and macOS.", file=sys.stderr)
+    if sys.platform not in {"darwin", "win32", "linux"}:
+        print("FCC Desktop is supported on Windows, macOS, and Linux.", file=sys.stderr)
         raise SystemExit(1)
 
-    from free_claude_code.cli.desktop_tray import launch as launch_tray
+    # Merge the FCC gateway routing block before any Claude Desktop session
+    # starts so model discovery and inference run through this server.
+    configure_claude_desktop_config()
```
[the remark this answers is not in the document]
Comment (Gakpey) [reply src/free_claude_code/cli/desktop_entrypoint.py]: Fixed in cbcf7cd. The entrypoint no longer calls `configure_claude_desktop_config()` at all — it runs before any lifecycle has started or verified a TLS front, so the merge there resolved the plain-HTTP fallback and wrote the reusable gateway credential against a cleartext listener. The routing merge now happens exclusively in the lifecycle layers, which merge only behind a verified HTTPS front (`desktop._merge_verified_gateway` gates on `verified_https_gateway_url()` and skips entirely when the front is unverified). `test_entrypoint_defers_desktop_routing_merge_to_the_lifecycle` proves no merge happens before the lifecycle starts.

[the remark this answers is not in the document]
Comment (Gakpey) [reply src/free_claude_code/cli/desktop_entrypoint.py]: Fixed in cbcf7cd — same fix as the sibling thread: the bare `configure_claude_desktop_config()` call was removed from the entrypoint entirely. Merging happens only in the lifecycle behind a verified HTTPS front (`desktop._merge_verified_gateway`), so the credential is never written pointing at the cleartext fallback. Regression guard: `test_entrypoint_defers_desktop_routing_merge_to_the_lifecycle`.

File: src/free_claude_code/cli/claude_desktop.py [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+        prior_inference = prior_inference if isinstance(prior_inference, dict) else {}
+        absorbed = dict(prior_inference)
+        for key in managed:
+            current_value = inference_dict.get(key)
+            if current_value == last_written.get(key):
+                continue  # Still FCC's written value; target unchanged.
+            absorbed[key] = current_value
+        if absorbed != prior_inference:
+            backup["inference"] = absorbed
+            changed = True
```
[the remark this answers is not in the document]
Comment (Gakpey) [reply src/free_claude_code/cli/claude_desktop.py:294]: Fixed in cbcf7cd. Two changes close this: (1) absorption now skips managed keys absent from the user's block — restoring a recorded target over an absent key would invent a literal `null` placeholder the user never wrote; (2) when the block became a mapping after the first merge recorded a non-object original verbatim (`inferenceRaw`), the re-merge now drops the stale verbatim target so unconfigure can no longer clobber the user's dict with the long-gone scalar. The per-key targets captured from the user's actual values are the only restore path. Regression guard: `test_remerge_after_scalar_to_dict_replacement_drops_raw_target` (scalar original → install → user replaces with own dict editing managed keys → re-merge → unconfigure → user's dict restored exactly, no null placeholders, no scalar restore).

File: src/free_claude_code/cli/tls_proxy.py [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+
+    path = front_identity_path(home_dir)
+    try:
+        existing = path.read_text(encoding="utf-8").strip()
+    except OSError:
+        existing = ""
+    if existing:
+        return existing
+
+    path.parent.mkdir(parents=True, exist_ok=True, mode=0o700)
```
[the remark this answers is not in the document]
Comment (Gakpey) [reply src/free_claude_code/cli/tls_proxy.py:151]: Fixed in ff6e7f7 ("fix: force owner-only permissions on an existing Caddy home directory", pushed to this branch).

`mkdir(mode=0o700)` only governs directories the call creates; a pre-existing home keeps its modes, so an inherited 0777 home let any local user traverse into the directory holding the identity secret, Caddyfile, and CA state. Both paths (`load_or_create_front_identity` and `CaddyTlsProxy.start`) now call `_restrict_home_permissions`, which chmods an existing directory to 0o700 — the same idiom `_save_config` already uses for the Claude Desktop config directory. Pinned by `test_load_or_create_front_identity_tightens_existing_permissive_home` and `test_caddy_start_tightens_existing_permissive_home` in tests/cli/test_tls_proxy.py.

**Comment by Gakpey** [comment]
Closing in favor of the v2 rebuild (see docs/superpowers/specs/2026-09-01-desktop-rerouting-v2-design.md on the fork). The simplified design replaces this stack: launch-flag TLS trust, spawn-only front, tray removal, and a compact config-engine rewrite that preserves all reviewer-mandated round-trip semantics. New PRs will reference this one.
